session/session_test.go

```diff
@@ -48,11 +52,64 @@ func TestReceiveEventWhenSubscribing(t *testing.T) {
 		if event != SessionDisconnected {
 			t.Error("Expected to receive disconnected: ", event)
 		}
+	case <-time.After(5 * time.Second):
+		t.Error("Failed to receive event")
+	}
 
-		if <-events != SessionReconnected {
+	select {
+	case event := <-events:
+		if event != SessionReconnected {
 			t.Error("Expected to receive reconnected: ", event)
 		}
-	case <-time.After(100 * time.Millisecond):
+	case <-time.After(5 * time.Second):
 		t.Error("Failed to receive event")
 	}
 }
+
+func TestResumeZKSessionWithValidSession(t *testing.T) {
+	proxy := CreateProxy(t)
+	defer proxy.Delete()
+
+	store, err := NewZKSession("localhost:27445", 12 * time.Second, nil)
+	if err != nil {
+		t.Error("Failed to connect to Zookeeper: ", err)
+	}
+	defer store.Close()
+
+	existingClientId, err := store.ClientId().Save()
+	if err != nil {
+		t.Error("Failed to save clientId: ", err)
+	}
+
+	store.quit <- "quit"
+
+	proxy.AddToxic("zookeeper_down", "timeout", "downstream", 1.0, toxiproxy.Attributes{
```

Comment:
I added this timeout toxic. The initial session is created with timeout of 12s and the toxic has a timeout of 6s but I still havent figured out the timing for the resume session to successfully reconnect (even though as I understand it there are 6s remaining in the initial session for it to reconnect).